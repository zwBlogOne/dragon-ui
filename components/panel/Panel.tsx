import React, { Component } from 'react';
import classnames from 'classnames';
import PropsType from './PropsType';

class Panel extends Component<PropsType, any> {
  static defaultProps = {
    theme: 'default',
    isRadius: false,
  };

  static Header;
  static Body;
  static Footer;
  static More;
  static Title;

  render() {
    const { props } = this;
    const {
      isRadius, theme, className, children, style,
    } = props;

    const cls = classnames({
      'ui-panel': true,
      radius: 'radius' in props || isRadius,
      [`theme-${theme}`]: !!theme,
      [className!]: !!className,
    });

    return (
      <div className={cls} style={style}>
        {children}
      </div>
    );
  }
}

export default Panel;
