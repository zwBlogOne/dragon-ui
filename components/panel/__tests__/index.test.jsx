import React from 'react';
import { render } from 'enzyme';
import toJson from 'enzyme-to-json';
import Panel from '../index';

describe('Panel', () => {
  it('renders Panel correctly', () => {
    const wrapper = render(
      <div>
        <Panel>
          <Panel.Header>
            <Panel.Title>头部左侧</Panel.Title>
            <Panel.More>
              头部右侧
            </Panel.More>
          </Panel.Header>
          <Panel.Body>
            React is the entry point to the React library
            If you load React from a script tag, these
            top-level APIs are available on the React global
          </Panel.Body>
          <Panel.Footer>
            <Panel.Title>底部左侧</Panel.Title>
            <Panel.More>
              底部右侧
            </Panel.More>
          </Panel.Footer>
        </Panel>
      </div>
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
