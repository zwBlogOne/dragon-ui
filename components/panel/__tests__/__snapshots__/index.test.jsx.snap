// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Panel renders Panel correctly 1`] = `
<div>
  <div
    class="ui-panel theme-default"
  >
    <div
      class="ui-panel-header"
    >
      <div
        class="ui-panel-title"
      >
        头部左侧
      </div>
      <div
        class="ui-panel-more"
      >
        头部右侧
      </div>
    </div>
    <div
      class="ui-panel-body"
    >
      React is the entry point to the React library If you load React from a script tag, these top-level APIs are available on the React global
    </div>
    <div
      class="ui-panel-footer"
    >
      <div
        class="ui-panel-title"
      >
        底部左侧
      </div>
      <div
        class="ui-panel-more"
      >
        底部右侧
      </div>
    </div>
  </div>
</div>
`;
